sim_dev_search/processors/tree_sitter/tree_sitter_extractor.py

```diff
@@ -0,0 +1,119 @@
+import json
+from collections import Counter
+from pathlib import Path
+from typing import List
+from urllib.parse import urlparse
+
+import git
+from tqdm import tqdm
+from tree_sitter import Language, Parser
+
+
+class TreeSitterExtractor:
+    """
+    Class that parses source code and extracts identifiers information.
+    """
+
+    BUILD_DIR = Path(__file__).resolve().parent / "build"
+    BUILD_LANGUAGES_PATH = BUILD_DIR / "my-languages.so"
+    CONFIG_PATH = Path(__file__).resolve().parent / "language_grammar_config.json"
+
+    IDENTIFIERS_QUERY = "(identifier) @variable"
```

Comment:
ideally you need to use language specific query - you can find it [here](https://github.com/JetBrains-Research/buckwheat/blob/371d272737bb45ece4bcd6eec07321f8797b44c6/identifiers_extractor/parsing.py#L63)